ts/src/kucoin.ts

```diff
@@ -780,6 +831,40 @@ export default class kucoin extends Exchange {
             const isDepositEnabled = this.safeValue (entry, 'isDepositEnabled', false);
             const fee = this.safeNumber (entry, 'withdrawalMinFee');
             const active = (isWithdrawEnabled && isDepositEnabled);
+            const networks = {};
+            const chains = this.safeValue (currencyChains, id, []);
+            for (let j = 0; j < chains.length; j++) {
+                const chain = chains[j];
+                const chainId = this.safeString (chain, 'chain');
+                const networkCode = this.networkIdToCode (chainId);
+                const chainName = this.safeString (chain, 'chainFullName');
+                const chainWithdrawEnabled = this.safeValue (chain, 'isWithdrawEnabled', false);
+                const chainDepositEnabled = this.safeValue (chain, 'isDepositEnabled', false);
+                const status = this.safeString (chain, 'status');
+                const chainEnabled = this.safeValue (chain, 'isChainEnabled', false);
+                const chainActive = chainEnabled && (status === 'enabled') && (chainWithdrawEnabled && chainDepositEnabled);
+                const chainWithdrawFee = this.safeNumber (chain, 'withdrawMinFee');
+                const chainPrecision = this.parseNumber (this.parsePrecision (this.safeString (chain, 'walletPrecision')));
+                const chainNetwork = {
+                    'id': chainId,
+                    'name': chainName,
+                    'code': networkCode,
+                    'active': chainActive,
+                    'fee': chainWithdrawFee,
+                    'precision': chainPrecision,
+                    'limits': {
+                        'withdraw': {
+                            'min': this.safeNumber (chain, 'withdrawMinSize'),
+                            'max': undefined,
+                        },
+                        'deposit': {
+                            'min': this.safeNumber (chain, 'depositMinSize'),
+                            'max': undefined,
+                        },
+                    },
+                };
+                networks[networkCode] = chainNetwork;
+            }
             result[code] = {
```

Comment:
yes, that's what I mentioned in top post. so, at first "networks unification" pr https://github.com/ccxt/ccxt/pull/16237  needs to be merged, then this will be shifted into working state

Reply:
@ttodua ok so I will try to merge the other one first

Reply:
@ttodua but what about the `usdt` value? it does not seem a network

Reply:
> @ttodua but what about the `usdt` value? it does not seem a network

hm,, let me retest and update with latest mergers included. will ping you

Reply:
I have revised them and you were correct,there were odd entries with "chain exists: false" that needed to be excluded from them, and I've added that change too, in addition to minor other touches. 
now it should work well.

Reply:
@ttodua will check again